asserting events

Hi,

I do this in my app

```
def mqtt_message_recieved_event(self, event_name, data, kwargs):  
        print(data)      
        payload = json.loads(data["payload"])
        message = ""
        for msg in payload["Messages"]:
            message += self.message_builder(msg)

        if message != "":
            self.fire_event("MQTT_PUBLISH", topic = "appdaemon/notification/tts", message = json.dumps({"message": message})) 
        else:
            self.log("Nothing to publish")

def initialize(self):
        self.log("NotitfyApp Init")
        self.set_namespace("mqtt")
        self.listen_event(self.mqtt_message_recieved_event, "MQTT_MESSAGE", topic = "homeassistant/notification/tts")
        self.set_namespace("hass")
```

How do I raise an event to test my `mqtt_message_recieved_event` method and assert that an event was fired?

Thanks

I have now got this in my test 
```
def test_callbacks_are_registered(given_that, notification_engine, assert_that):
    assert_that(notification_engine) \
        .listens_to.event('MQTT_MESSAGE', topic = "homeassistant/notification/tts") \
        .with_callback(notification_engine.mqtt_message_recieved_event)
    
def test_during_night_light_turn_on(given_that, notification_engine, assert_that):
    given_that.state_of('light').is_set_to(light_states)    
    given_that.state_of('timer').is_set_to(timer_states)
    notification_engine.mqtt_message_recieved_event(None, {'topic': 'homeassistant/notification/tts', 'payload': '{"Messages":["Windows","Doors","LightsCount","Tele"]}'},  None)
```
However I'm not sure how to assert the event got fired and I'm getting an error
```
=================================================================================================== FAILURES ====================================================================================================
______________________________________________________________________________ test_during_night_light_turn_on[NotificationEngine] ______________________________________________________________________________

given_that = <appdaemontestframework.given_that.GivenThatWrapper object at 0xb54cae30>, notification_engine = <apps.NotificationEngine.NotificationEngine object at 0xb53ce130>
assert_that = <appdaemontestframework.assert_that.AssertThatWrapper object at 0xb53ce110>

    def test_during_night_light_turn_on(given_that, notification_engine, assert_that):
        given_that.state_of('light').is_set_to(light_states)
        given_that.state_of('timer').is_set_to(timer_states)
>       notification_engine.mqtt_message_recieved_event(None, {'topic': 'homeassistant/notification/tts', 'payload': '{"Messages":["Windows","Doors","LightsCount","Tele"]}'},  None)

test/NotificationEngineTest.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
apps/NotificationEngine.py:19: in mqtt_message_recieved_event
    self.fire_event("MQTT_PUBLISH", topic = "appdaemon/notification/tts", message = json.dumps({"message": message}))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <apps.NotificationEngine.NotificationEngine object at 0xb53ce130>, args = ('MQTT_PUBLISH',), kwargs = {'message': '{"message": "1 light on"}', 'topic': 'appdaemon/notification/tts'}, is_async = True
f = <Task pending coro=<fire_event() running at /usr/local/lib/python3.6/site-packages/appdaemon/adapi.py:1666>>

    @wraps(coro)
    def inner_sync_wrapper(self, *args, **kwargs):
        is_async = None
        try:
            # do this first to get the exception
            # otherwise the coro could be started and never awaited
            asyncio.get_event_loop()
            is_async = True
        except RuntimeError:
            is_async = False
    
        if is_async is True:
            # don't use create_task. It's python3.7 only
            f = asyncio.ensure_future(coro(self, *args, **kwargs))
>           self.AD.futures.add_future(self.name, f)
E           AttributeError: 'NotificationEngine' object has no attribute 'AD'

/usr/local/lib/python3.6/site-packages/appdaemon/utils.py:189: AttributeError
--------------------------------------------------------------------------------------------- Captured stdout call ----------------------------------------------------------------------------------------------
{'topic': 'homeassistant/notification/tts', 'payload': '{"Messages":["Windows","Doors","LightsCount","Tele"]}'}
========================================================================================== 1 failed, 1 passed in 0.33s ==========================================================================================
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,7 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 *
 
 ## Fixes
-*
+* Automations can fire events without crashing the test framework
 
 ## Breaking Changes
 * None

--- a/appdaemontestframework/hass_mocks.py
+++ b/appdaemontestframework/hass_mocks.py
@@ -93,6 +93,7 @@ class HassMocks:
             MockHandler(Hass, 'call_service'),
             MockHandler(Hass, 'turn_on'),
             MockHandler(Hass, 'turn_off'),
+            MockHandler(Hass, 'fire_event'),
 
             ### Custom callback functions
             MockHandler(Hass, 'register_constraint'),

--- a/test/test_events.py
+++ b/test/test_events.py
@@ -1,0 +1,27 @@
+from appdaemon.plugins.hass.hassapi import Hass
+
+from appdaemontestframework import automation_fixture
+
+LIGHT = 'light.some_light'
+COVER = 'cover.some_cover'
+
+
+class MockAutomation(Hass):
+    def initialize(self):
+        pass
+
+    def send_event(self):
+        self.fire_event("SOME_EVENT", my_keyword="hello")
+
+
+@automation_fixture(MockAutomation)
+def automation():
+    pass
+
+
+def test_it_does_not_crash_when_testing_automation_that_sends_events(given_that,
+                                                                     automation: MockAutomation):
+    # For now, there is not assertion feature on events, so we're just ensuring
+    # appdaemontestframework is not crashing when testing an automation that
+    # sends events.
+    automation.send_event()

